roles/kubernetes-apps/ingress_controller/ambassador/templates/clusterrole-ambassador.yml.j2

```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: ambassador-operator-cluster
+  labels:
+    app.kubernetes.io/name: ambassador-operator
+    app.kubernetes.io/part-of: ambassador-operator
+rules:
+  - apiGroups: ['*']
```

Comment:
Can these perms be a bit more restrictive ?

Reply:
We could try to tighten it based on the resources that are currently installed by the Ambassador Operator. This is a relatively long list of resources, so the permissions need to be really wide, although they could be reduced.

However, the main point of concern is that, since this operator installs the latest version of Ambassador _and keeps it up to date over time_, these permissions could go stale and cause unexpected errors for our users in the future, when some extra resources could need to be installed. So that is the reason we've kept the permissions intentionally relaxed.

There are other operators (specifically which perform upgrades) in the wild which do the same - one example would be https://github.com/fluxcd/helm-operator/blob/157d422bc11f127b1d9a5eba18fc8882c4b9cde9/deploy/rbac.yaml#L10-L21.